crypto/hrss/hrss.c

```diff
@@ -914,20 +913,28 @@ void HRSS_poly3_invert(struct poly3 *out, const struct poly3 *in) {
 // Coefficients are ordered little-endian, thus the coefficient of x^0 is the
 // first element of the array.
 struct poly {
-#if defined(HRSS_HAVE_VECTOR_UNIT)
-  union {
-    // N + 3 = 704, which is a multiple of 64 and thus aligns things, esp for
-    // the vector code.
-    uint16_t v[N + 3];
-    vec_t vectors[VECS_PER_POLY];
-  };
-#else
+  // TODO(shang): investigate below comment. alignas(16) is c11 extension.
   // Even if !HRSS_HAVE_VECTOR_UNIT, external assembly may be called that
   // requires alignment.
-  alignas(16) uint16_t v[N + 3];
-#endif
+  uint16_t v[N + 3];
 };
 
+#if defined(HRSS_HAVE_VECTOR_UNIT)
+struct poly_vec {
+    vec_t vectors[VECS_PER_POLY];
```

Comment:
You have to align this array because the code below uses SSE instructions (that's why previously there was `alignas(16)`.